src/vmm/src/devices/virtio/virtio_block/device.rs

```diff
@@ -66,24 +66,26 @@ impl FileEngineType {
 pub struct DiskProperties {
     pub file_path: String,
     pub file_engine: FileEngine<PendingRequest>,
+    pub is_disk_read_only: bool,
```

Comment:
We already have `read_only` in the block itself. I think we don't need to duplicate it here.

Reply:
I think this is actually does not belong here. I think this is not a big deal to pass it to `update` method with `disk_image_path`:
```rust
pub fn update(&mut self, disk_image_path: String, read_only: bool) -> Result<(), VirtioBlockError> {
  ...
}

self.disk.update(disk_image_path, self.read_only)?;
```

Reply:
I disagree. I don't think it makes sense to pass this over, since the PATCH command does not allow changing the read/write semantics of the file.

Reply:
This is for the `VirtioBlock` type to decide what can be changed. The `DiskProperties` is just a helper fro building `image_id` and `virtio_config_space`. It does not need to know if disk it `readonly` or not. The `readonly` property is only needed for `file_engine`, but we discussed that it can be moved out. 
Anyway, maybe this can be done as a part of the refactoring.

Reply:
Ok, I moved the `is_read_only` thing out of `DiskProperties` ;) We can think of how to organize this better if/when we refactor.